RFS/src/main/java/org/opensearch/migrations/bulkload/common/ByteArrayIndexInput.java

```diff
@@ -3,7 +3,7 @@
 import java.io.EOFException;
 import java.io.IOException;
 
-import org.apache.lucene.store.IndexInput;
+import shadow.lucene9.org.apache.lucene.store.IndexInput;
```

Comment:
What are your thoughts on how to handle when support for Lucene 10/OS 3.X becomes a requirement?  What if instead of having `lucene9` to represent the "latest" Lucene version, we do something like `lucenelatest`?  That way, we can just swap the version number in gradle and not have to update all our imports.  If it turns out we need to explicitly specify a version here, that's a whole different can of worms.

Reply:
Per conversation w/ @peternied, the plan is to assume all of these classes are inherently versioned and begin moving them to the new `org.opensearch.migrations.bulkload.lucene` package when that becomes an issue.  E.g. we'll lazily resolve that issue.

Reply:
I'd rather we are explicit about which version of lucene is used, this should be trivial since the source versions is expected before we read a snapshot.  For some of these lingering places where we have lucene references, I think we should cleanup/consolidate them.